src/Diabetes.Core/Models/Food.cs

```diff
@@ -48,5 +48,16 @@ public PropertyValue ValidateNewValue<PropertyValue>(PropertyValue value, string
                 throw new InvalidDataException(message);
             return value;
         }
+
+        public MeasuredAmount ValidateNewValue(MeasuredAmount value, string message)
```

Comment:
We need to figure out how to do validation messages that are dynamic based on what is actually wrong.